### DISCUSSION TITLE
FCC Detectors

### DISCUSSION THREAD

**Discussion by kjvbrt** [HSF/phoenix#458] [Q&A] [answered]
Hello,

following brief discussion at the Wednesday meeting, we would like to visualize FCC Detectors and simulated events using Phoenix.
First attempt is published here:
https://kjvbrt.github.io/FCCSW/
and one can have a look at the code here:
https://raw.githubusercontent.com/kjvbrt/FCCSW/phoenix/geo/geo-phoenix.md

Currently there are no event data.
I'm not sure about the interface buttons, but they are not showing.
The end goal is to have a tool which can easily visualize detector geometry (+ event data) for someone who is working on detector design.

![image](https://user-images.githubusercontent.com/34742917/172910305-f426f548-39df-4fb6-ba9d-7d2f8ebfe523.png)

**Comment by 9inpachi** [comment]
Hi @kjvbrt,

Since you are using Phoenix as a standalone bundle, this doesn't have any UI components.

The UI menu (icon bar) and Phoenix menu are Angular components part of the `phoenix-ui-components` package that can only be used inside an Angular application. We have a dedicated guide for how to set up Phoenix for a new experiment by creating an Angular application. You can read it here: https://github.com/HSF/phoenix/blob/master/guides/developers/set-up-phoenix.md

You can enable the dat.GUI menu as an alternative to Phoenix menu if you are using the standalone bundle but it doesn't have as many options as the Phoenix menu.

You can enable it using:

```js
// Define the configuration
const configuration = {
  elementId: 'eventDisplay',
  eventDataLoader: new PhoenixLoader(),
  enableDatGUIMenu: true
};
```

**Comment by kjvbrt** [comment]
Hello @9inpachi,

thanks for the pointers :)
I set-up a testing angular application here:
https://kjvbrt.github.io/fcc-viewer/

Since FCC detectors are in the development phase, we need to be able to load detector geometry and event data from the local machine of the user. Our workflow is expected to look something like this:
User is connected to lxplus with key4hep stack loaded. After making changes in FCCDetectors he/she will recompile the package and generate few events. With local versions of the compact file and file containing the events he/she will start launcher of the phoenix which in turn will convert everything into the appropriate formats and opens browser window with the outcome.

I'm thinking of splitting the application into two parts. One will do the conversions and will serve the converted files (launcher), the other part (angular phoenix application) will be hosted on the remote server and will visualize the detector.

What do you think of architecture like this? Do you see any short/longterm problems?

**Comment by 9inpachi** [comment, marked as answer]
Hi @kjvbrt,

Apologies, I completely missed your message! :(

I am not familiar with the different stacks at CERN but I think the architecture you have in mind should work. What I understand is that you will use the launcher to serve the events through a URL and then the Phoenix application will consume the URL to show these events.

It might be enough that the launcher just serves the files and doesn't convert them. We have a guide on creating [event data loaders](https://github.com/HSF/phoenix/blob/master/guides/developers/event-data-loader.md) which can be used to implement an event data converter from your native experiment format to the Phoenix friendly format directly inside the Phoenix application. So the application can consume an unfriendly format and use the event data loader to convert and display the event data.

But if you want the converter and the application to scale independently or if the conversion is relatively complex, then it makes sense to do the conversion using the launcher.

I hope this helps.
